### PR TITLE
hw5: add mongo connection, basic endpoints\models

### PR DESCRIPTION
**Pull request by ValeriyPetryna**
Init mongo connection, add new endpoints, init mongoose schemas, minor changes

### CODE REVIEW

File: api/db/models/blog.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    content: {
+      required: true,
+      type: String,
+    },
+    likes: {
+      type: [String],
+      default: [],
+    },
+    comments: {
+      ref: 'comment',
```
Comment (afandul) [inline api/db/models/blog.js:44]: Reference should correspond to entity name, stated in L53 as 'Comment'

File: api/db/models/user.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    default: Date.now,
+  },
+  updatedAt: {
+    type: String,
+    type: Date,
+    default: Date.now,
+  },
+});
+
+const User = mongoose.model("Users", user);
```
Comment (afandul) [inline api/db/models/user.js:38]: Entity name shouldn't be in plural like 'Users', use 'User' instead

File: api/db/mongo.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+const mongoose = require("mongoose");
+
+const { MONGO_URL } = require("../utils/config");
+
+module.exports = () => {
+  mongoose
+    .connect(MONGO_URL, {
+      dbName: "blog",
```
Comment (afandul) [inline api/db/mongo.js:8]: Better keep db name in uri

File: api/posts/controller.js [hunk 4]
```diff
@@ -1,55 +1,88 @@
...
+    res.send(posts);
+  } catch (error) {
+    next(error);
+  }
+};
+
+const getSinglePost = async (req, res, next) => {
+  try {
+    const id = req.params.id;
+    const post = await Post.findById(id).exec();
```
Comment (afandul) [inline api/posts/controller.js:16]: exec() is redundant

**Review by afandul — APPROVED** [review]
Good solution, pay attention to refs in schemas

File: api/posts/controller.js (hunk 4 above)
Comment (ValeriyPetryna) [reply api/posts/controller.js:16]: exec removed in the next PR

File: api/db/mongo.js (hunk 3 above)
Comment (ValeriyPetryna) [reply api/db/mongo.js:8]: added to uri, thanks!

File: api/db/models/user.js (hunk 2 above)
Comment (ValeriyPetryna) [reply api/db/models/user.js:38]: fixed, thanks

File: api/db/models/blog.js (hunk 1 above)
Comment (ValeriyPetryna) [reply api/db/models/blog.js:44]: fixed

**Pull request merged**